sklearn/preprocessing/target.py

```diff
@@ -0,0 +1,224 @@
+                check_inverse=self.check_inverse)
+        # XXX: sample_weight is not currently passed to the
+        # transformer. However, if transformer starts using sample_weight, the
+        # code should be modified accordingly. At the time to consider the
+        # sample_prop feature, it is also a good use case to be considered.
+        self.transformer_.fit(y)
+        if self.check_inverse:
+            idx_selected = slice(None, None, max(1, y.shape[0] // 10))
+            if not np.allclose(
+                    safe_indexing(y, idx_selected),
+                    self.transformer_.inverse_transform(
+                        self.transformer_.transform(
+                            safe_indexing(y, idx_selected)))):
+                warnings.warn("The provided functions or transformer are"
+                              " not strictly inverse of each other. If"
+                              " you are sure you want to proceed regardless"
+                              ", set 'check_inverse=False'", UserWarning)
+
+    def fit(self, X, y, sample_weight=None):
+        """Fit the model according to the given training data.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape (n_samples, n_features)
+            Training vector, where n_samples is the number of samples and
+            n_features is the number of features.
+
+        y : array-like, shape (n_samples,)
+            Target values.
+
+        sample_weight : array-like, shape (n_samples,) optional
+            Array of weights that are assigned to individual samples.
+            If not provided, then each sample is given unit weight.
+
+        Returns
+        -------
+        self : object
+            Returns self.
+        """
+        y = check_array(y, accept_sparse='csr', force_all_finite=True,
+                        ensure_2d=False, dtype='numeric')
+
+        # store the number of dimension of the target to predict an array of
+        # similar shape at predict
+        self.training_dim_ = y.ndim
+
+        # transformers are designed to modify X which is a 2d dimensional, we
+        # need to modify y accordingly.
+        if y.ndim == 1:
+            y_2d = y.reshape(-1, 1)
+        else:
+            y_2d = y
+        self._fit_transformer(y_2d)
+
+        if self.regressor is None:
+            from ..linear_model import LinearRegression
+            self.regressor_ = LinearRegression()
+        else:
+            self.regressor_ = clone(self.regressor)
+
+        # transform y and convert back to 1d array if needed
+        y_trans = self.transformer_.fit_transform(y_2d)
+        # FIXME: a FunctionTransformer can return a 1D array even when validate
+        # is set to True. Therefore, we need to check the number of dimension
+        # first.
+        if y_trans.ndim == 2 and y_trans.shape[1] == 1:
+            y_trans = y_trans.squeeze(axis=1)
```

Comment:
You have accepted sparse input above, but sparse won't work here. (I don't think we need to accept sparse input though...)